coordinator/src/pyroscope.py

```diff
@@ -66,11 +77,17 @@ def _build_store_gateway_config(self, roles_addresses: Dict[str, Set[str]]):
             )
         )
 
-    def _build_memberlist_config(self, worker_peers: Optional[Tuple[str, ...]]):
-        return pyroscope_config.Memberlist(
+    def _build_memberlist_config(self, worker_peers: Optional[Tuple[str, ...]], tls = False):
+        memberlist_config = pyroscope_config.Memberlist(
             bind_port=self.memberlist_port,
             join_members=([f"{peer}:{self.memberlist_port}" for peer in worker_peers] if worker_peers else []),
         )
+        if tls:
```

Comment:
same as above. I usually do:

```suggestion
        tls_config = {...} if tls else {}
        return pyroscope_config.Memberlist(..., **tls_config)
```